### PR TITLE
feat(map): improve map UX and production readiness

### PR DESCRIPTION
**Pull request by jansinger**
## Summary

- **Global view**: Remove Baltic Sea extent constraint and lower minZoom to 2, allowing users to zoom out to world level
- **Co-located cluster list**: Show scrollable sighting list with detail navigation for features at the same coordinates, replacing the unhelpful "zoom in" message
- **No-results hints**: Display contextual messages when no sightings exist for the selected year or when all sightings are hidden by filters
- **Hover/popup overlap fix**: Hide hover tooltip when popup is open and on click
- **Production-readiness fixes**: CSP-safe CSS hover, ARIA labels, parseInt validation, timer cleanup on dispose, keyboard handler edge cases

## Test plan

- [ ] Verify map can zoom out to world level and back
- [ ] Switch to a year with no data (e.g. 2010) — confirm "Keine Sichtungen" hint appears
- [ ] Use time slider to filter out all sightings — confirm filter hint appears
- [ ] Click a cluster with multiple sightings at the same location — confirm scrollable list appears
- [ ] Click a list entry — confirm detail view with back button works
- [ ] Hover over marker, then click it — confirm hover disappears and popup shows cleanly
- [ ] Tab through cluster list items — confirm focus-visible styling
- [ ] Run `npm run test:quick` — all checks pass (1219 unit tests)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/map/optimizedMapController.ts [hunk 1]
```diff
@@ -476,38 +495,122 @@ export class SichtungenMap {
...
-		content += `
-				<div style="margin-top: 10px; padding-top: 8px; border-top: 1px solid #eee; font-size: 12px; color: #666;">
-					Zoomen Sie hinein für Details
+		return `
+			<div class="cluster-popup" role="list" aria-label="${count} Sichtungen an diesem Ort">
+				<h3 style="margin: 0 0 8px 0; color: #333; font-size: 14px;">${count} Sichtungen an diesem Ort</h3>
+				<div style="max-height: 240px; overflow-y: auto; padding-right: 4px;">
+					${items}
 				</div>
 			</div>
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:559]: Fixed in 61e3fac: Replaced `role="list"` on `<div>` with semantic `<ul>/<li>` structure. Each `<button>` is now wrapped in a `<li>`, and the container uses `<ul aria-label="...">`.

File: src/lib/map/optimizedMapController.ts [hunk 2]
```diff
@@ -573,6 +676,8 @@ export class SichtungenMap {
 			// Redraw und Zeitraum-Anzeige aktualisieren nachdem timeFilter gesetzt wurde
 			this.reportsLayer.changed();
 			this.updateTimeRange();
+			// Counts mit korrektem timeFilter neu berechnen (loadSightings hat sie mit dem alten berechnet)
+			this.legendUpdateCallback?.();
 			this.loadingCallback?.(false);
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:668]: Fixed in 61e3fac: Removed `legendUpdateCallback` from `loadSightings()`. Callers (`setYear`, `applyFilter`) are now solely responsible for triggering count updates after setting the correct `timeFilter`, eliminating the duplicate computation.

File: src/lib/map/optimizedMapController.ts [hunk 3]
```diff
@@ -714,10 +820,10 @@ export class SichtungenMap {
 	private applyFilter(searchTerm: string): void {
 		this.searchTerm = searchTerm;
 		this.loadingCallback?.(true);
-		// fire-and-forget: unhandled rejection propagiert als window.unhandledrejection
-		// und wird vom Error-Handler in SightingsMapView aufgefangen
 		void this.loadSightings(this.displayedYear, searchTerm)
 			.then(() => {
+				this.reportsLayer.changed();
+				this.legendUpdateCallback?.();
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:812]: Fixed in 61e3fac: Same fix — `legendUpdateCallback` removed from `loadSightings()`. `applyFilter()` calls it once after `reportsLayer.changed()`, no more double computation.

File: src/lib/components/map/SightingsMapView.svelte [hunk 4]
```diff
@@ -83,6 +83,21 @@
...
+	// Feature-Anzahl direkt vom Map-Controller abfragen (robuster als counts-basiert)
+	let featureCount = $state(0);
+	let totalFeatures = $derived(
+		Object.values(counts.speciesCounts).reduce((sum, c) => sum + c.total, 0)
+	);
+	let visibleFeatures = $derived(
+		Object.values(counts.speciesCounts).reduce((sum, c) => sum + c.visible, 0)
+	);
+	let showNoResults = $derived(
+		!isInitialLoading && !isLoadingData && featureCount === 0 && totalFeatures === 0
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/components/map/SightingsMapView.svelte]: Fixed in 61e3fac: Added `!errorMessage` guard to `showNoResults`. No-results hint is now suppressed when an error toast is visible.

File: src/lib/components/map/SightingsMapView.svelte [hunk 5]
```diff
@@ -83,6 +83,21 @@
...
+	);
+	let showNoVisibleResults = $derived(
+		!isInitialLoading && !isLoadingData && featureCount > 0 && visibleFeatures === 0
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/components/map/SightingsMapView.svelte]: Fixed in 61e3fac: Added `!errorMessage` guard to `showNoVisibleResults` as well.

File: src/lib/map/optimizedMapController.ts [hunk 6]
```diff
@@ -476,38 +495,122 @@ export class SichtungenMap {
...
+		return features.every((f) => {
+			const geom = f.getGeometry();
+			if (!geom) return true;
+			const ext = geom.getExtent() as [number, number, number, number];
+			return (
+				Math.abs(ext[0] - firstExtent[0]) < threshold &&
+				Math.abs(ext[1] - firstExtent[1]) < threshold
+			);
+		});
+	}
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:517]: Fixed in 61e3fac: Extracted `areExtentsColocated()` as a pure function into `mapUtils.ts` and added 7 unit tests covering: empty/single input, identical coords, within/beyond threshold, Y-axis only deviation, default threshold, and mixed distances with outlier.

File: src/lib/map/optimizedMapController.ts [hunk 7]
```diff
@@ -267,6 +263,17 @@ export class SichtungenMap {
...
+					padding: 8px; margin-bottom: 4px; border: 1px solid #e5e7eb;
+					border-radius: 6px; background: #f9fafb; cursor: pointer;
+					text-align: left; font-size: 13px; transition: background 0.15s;
+				}
+				.cluster-list-item:hover, .cluster-list-item:focus-visible {
+					background: #e0f2fe; outline: 2px solid #2563eb; outline-offset: -2px;
+				}
+			</style>
 			<div class="ol-popup-content">
 				<button class="ol-popup-closer" type="button">×</button>
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts]: Fixed in dc37d68: Added `aria-label="Popup schließen"` to the close button.

File: src/lib/map/optimizedMapController.ts [hunk 8]
```diff
@@ -476,38 +495,112 @@ export class SichtungenMap {
...
+	private attachClusterListHandlers(
+		contentDiv: HTMLDivElement,
+		features: Feature<Geometry>[],
+		coordinate: number[]
+	): void {
+		const sorted = [...features].sort((a, b) => {
+			const tsA = (a.getProperties() as SightingProperties).ts || 0;
+			const tsB = (b.getProperties() as SightingProperties).ts || 0;
+			return tsB - tsA;
+		});
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts]: Fixed in dc37d68: Extracted `sortFeaturesByDate()` helper. Sort happens once in `showPopup()`, and the sorted array is passed to both `createClusterListContent()` and `attachClusterListHandlers()`.

File: src/lib/components/map/Panel/FilterPanel.svelte [hunk 9]
```diff
@@ -30,18 +30,20 @@
...
 		isApplyingFilter = true;
-
-		// Simuliere Filterzeit (wird durch den echten Filter-Prozess ersetzt)
-		setTimeout(() => {
+		if (filterFeedbackTimeout) clearTimeout(filterFeedbackTimeout);
+		filterFeedbackTimeout = setTimeout(() => {
 			isApplyingFilter = false;
-		}, 1500);
+			filterFeedbackTimeout = null;
+		}, 800);
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/components/map/Panel/FilterPanel.svelte:41]: Fixed in dc37d68: Added `$effect` cleanup that calls `clearTimeout(filterFeedbackTimeout)` on unmount.

File: src/lib/map/optimizedMapController.ts [hunk 10]
```diff
@@ -318,6 +325,13 @@ export class SichtungenMap {
 		if (!isTouchDevice) {
 			this.map.on('pointermove', (event) => {
 				const infoElement = document.getElementById('info');
+
+				// Hover ausblenden wenn ein Popup offen ist
+				if (this.popup.getPosition()) {
+					if (infoElement) infoElement.style.display = 'none';
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:331]: Fixed in dc37d68: Added `this.map.getTargetElement().style.cursor = ''` in the early return path when popup is open.
